controllers/topology/cluster_class.go

```diff
@@ -77,6 +77,8 @@ func (r *ClusterReconciler) getClass(ctx context.Context, cluster *clusterv1.Clu
 	for _, mdc := range class.clusterClass.Spec.Workers.MachineDeployments {
 		mdTopologyClass := machineDeploymentTopologyClass{}
 
+		mdc.Template.Metadata.DeepCopyInto(&mdTopologyClass.metadata)
```

Comment:
Nice finding! I think we have the same problem with controlPlane.metadata. @killianmuldoon Probably makes sense if you add similar code for control plane in your PR (if not already done)

Reply:
I'm not seeing the impact this has - can you explain what this change is addressing?

Reply:
In my getClass PR I forgot to read the metadata from the ClusterClass into the `clusterTopologyClass` struct. Thus it has been never read, never computed and also cannot be reconciled.

Just realized now that your PR is not the compute but the reconcile part.

Anyway, we should also read / compute / reconcile controlPlane metadata. We can either add that in our PR or open an issue to add it afterwards. WDYT?

Reply:
Hm, took another look at the code of `computeControlPlane`. Looks like there's nothing to do. `computeControlPlane` reads the controlPlane labels directly from the ClusterClass. 

I initially thought when we copy metadata for MachineDeployments we also should do it for the controlPlane. But I guess the difference is that for controlPlane it's easy to just read it from the ClusterClass while it's a bit more complicated (MachineDeployment topology name matching) for MDs, which is why we're copying it here.


tl;dr nothing to do, if I'm right

Reply:
totally makes sense, should have caught that earlier. Resolving the thread.

Reply:
Please add a comment on top of it to explain why this is here and its scope.